refactor: Widget Development API (non-connected widgets)

riodeuno opened the pull request:
## Description

Fixes #5057


## Type of change
- Structural changes (non-breaking change which does not add functionality)

## How Has This Been Tested?

- Changes are valid if existing tests pass

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
















## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>


    // Code coverage diff between base branch:release and head branch: feature/widget-development-api 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.38 **(0.08)** | 36.25 **(-0.08)** | 33.58 **(0.13)** | 54.94 **(0.09)**
 :green_circle: | app/client/src/comments/utils.ts | 92.98 **(2.07)** | 50 **(3.33)** | 70 **(0)** | 91.49 **(2.6)**
 :sparkles: :new: | **app/client/src/components/constants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/BaseButton.tsx** | **27.03** | **10.26** | **0** | **32.26**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/CloseButton.tsx** | **55.56** | **100** | **0** | **55.56**
 :green_circle: | app/client/src/components/designSystems/appsmith/PositionedContainer.tsx | 100 **(0)** | 87.5 **(4.17)** | 100 **(0)** | 100 **(0)**
 :green_circle: | app/client/src/components/editorComponents/ApiResponseView.tsx | 51.81 **(0.59)** | 38.96 **(1.63)** | 0 **(0)** | 53.09 **(0.59)**
 :green_circle: | app/client/src/components/editorComponents/DropTargetComponent.tsx | 89.74 **(0.27)** | 67.92 **(2.61)** | 90.91 **(0)** | 89.47 **(0.28)**
 :red_circle: | app/client/src/components/editorComponents/ActionRightPane/SuggestedWidgets.tsx | 41.3 **(-1.25)** | 26.67 **(0)** | 0 **(0)** | 42.22 **(-1.26)**
 :green_circle: | app/client/src/components/editorComponents/WidgetNameComponent/index.tsx | 95.45 **(0.07)** | 80.7 **(0)** | 100 **(0)** | 95.31 **(0.07)**
 :red_circle: | app/client/src/components/propertyControls/BorderRadiusOptionsControl.tsx | 53.57 **(-4.49)** | 25 **(-8.33)** | 25 **(-8.33)** | 58.33 **(-4.63)**
 :red_circle: | app/client/src/components/propertyControls/BoxShadowOptionsControl.tsx | 57.14 **(-7.57)** | 33.33 **(-9.53)** | 25 **(-8.33)** | 60 **(-7.74)**
 :red_circle: | app/client/src/entities/Widget/utils.ts | 96.97 **(-0.04)** | 85.29 **(0)** | 100 **(0)** | 96.88 **(-0.04)**
 :red_circle: | app/client/src/icons/WidgetIcons.tsx | 62.12 **(-33.33)** | 100 **(0)** | 21.88 **(-68.75)** | 62.12 **(-33.33)**
 :green_circle: | app/client/src/pages/Editor/EditorHeader.tsx | 89.72 **(0.2)** | 55.56 **(0)** | 69.23 **(0)** | 89.32 **(0.21)**
 :green_circle: | app/client/src/pages/Editor/PropertyPaneHelpButton.tsx | 85 **(0)** | 75 **(33.33)** | 50 **(0)** | 84.21 **(0)**
 :green_circle: | app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx | 77.59 **(0.2)** | 68 **(0)** | 68.97 **(0)** | 77.57 **(0.21)**
 :green_circle: | app/client/src/pages/Editor/APIEditor/index.tsx | 31.43 **(0.66)** | 3.64 **(0)** | 5 **(0)** | 34.04 **(0.71)**
 :red_circle: | app/client/src/pages/Editor/Explorer/EntityExplorer.tsx | 84 **(-0.31)** | 40 **(0)** | 50 **(0)** | 85.11 **(-0.31)**
 :green_circle: | app/client/src/pages/Editor/Explorer/Entity/Name.tsx | 59.74 **(0.53)** | 26.83 **(0)** | 46.15 **(0)** | 58.9 **(0.57)**
 :red_circle: | app/client/src/pages/Editor/Explorer/Pages/PageEntity.tsx | 90.63 **(-0.28)** | 40 **(0)** | 60 **(0)** | 90.63 **(-0.28)**
 :green_circle: | app/client/src/pages/Editor/Explorer/Widgets/WidgetContextMenu.tsx | 75.61 **(1.93)** | 16.67 **(0)** | 66.67 **(0)** | 75 **(2.03)**
 :red_circle: | app/client/src/pages/Editor/Explorer/Widgets/WidgetEntity.tsx | 97.92 **(-0.08)** | 82.98 **(0)** | 100 **(0)** | 100 **(0)**
 :green_circle: | app/client/src/pages/Editor/Explorer/Widgets/useNavigateToWidget.ts | 90.7 **(0.22)** | 75 **(0)** | 100 **(0)** | 92.68 **(0.18)**
 :red_circle: | app/client/src/pages/Editor/PropertyPane/ConnectDataCTA.tsx | 40.54 **(-3.05)** | 100 **(0)** | 7.14 **(0)** | 38.89 **(-3.22)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/index.tsx | 83.33 **(0.14)** | 69.3 **(0)** | 70.37 **(0)** | 85.05 **(0.29)**
 :green_circle: | app/client/src/reducers/entityReducers/widgetConfigReducer.tsx | 100 **(14.29)** | 100 **(0)** | 100 **(100)** | 100 **(14.29)**
 :green_circle: | app/client/src/sagas/ModalSagas.ts | 49.21 **(1.67)** | 25 **(0)** | 30 **(0)** | 50 **(1.79)**
 :green_circle: | app/client/src/sagas/OnboardingSagas.ts | 17.43 **(0.45)** | 1.46 **(0)** | 12.24 **(0)** | 17.67 **(0.53)**
 :green_circle: | app/client/src/sagas/PageSagas.tsx | 24.07 **(0.47)** | 6.25 **(0)** | 14.29 **(0)** | 25.81 **(0.54)**
 :green_circle: | app/client/src/sagas/SnipingModeSagas.ts | 22 **(1.59)** | 3.33 **(0)** | 40 **(0)** | 22.92 **(1.64)**
 :red_circle: | app/client/src/sagas/WidgetBlueprintSagas.ts | 58.18 **(-0.75)** | 37.5 **(0)** | 44.44 **(0)** | 55.1 **(-0.9)**
 :red_circle: | app/client/src/sagas/WidgetEnhancementHelpers.ts | 36.99 **(0)** | 25.49 **(-2.17)** | 14.29 **(0)** | 39.39 **(-0.61)**
 :red_circle: | app/client/src/sagas/WidgetOperationSagas.tsx | 52.33 **(-1.57)** | 37.38 **(0.55)** | 51.95 **(-2)** | 53.02 **(-1.21)**
 :green_circle: | app/client/src/sagas/WidgetSelectionSagas.ts | 91.28 **(0.18)** | 87.1 **(0)** | 100 **(0)** | 90.63 **(0.23)**
 :red_circle: | app/client/src/sagas/index.tsx | 92.31 **(-0.14)** | 60 **(0)** | 100 **(0)** | 94 **(-0.12)**
 :green_circle: | app/client/src/sagas/selectors.tsx | 76 **(2.36)** | 62.5 **(4.81)** | 54.84 **(0.29)** | 71.05 **(3.19)**
 :red_circle: | app/client/src/selectors/editorSelectors.tsx | 69.88 **(-0.85)** | 57.63 **(-2.02)** | 52.73 **(-2.83)** | 67.15 **(-1)**
 :red_circle: | app/client/src/selectors/themeSelectors.tsx | 95.45 **(1.01)** | 66.67 **(-33.33)** | 83.33 **(3.33)** | 93.75 **(1.44)**
 :green_circle: | app/client/src/selectors/widgetSelectors.ts | 87.5 **(-0.74)** | 62.5 **(12.5)** | 80 **(-3.33)** | 92.31 **(-0.55)**
 :sparkles: :new: | **app/client/src/utils/DSLMigrations.ts** | **74.73** | **66.33** | **69.77** | **74.32**
 :green_circle: | app/client/src/utils/WidgetFactory.tsx | 73.63 **(1.87)** | 70.59 **(1.84)** | 64.71 **(2.21)** | 73.63 **(1.87)**
 :green_circle: | app/client/src/utils/WidgetPropsUtils.tsx | 86.9 **(11.01)** | 74 **(7.09)** | 84.62 **(12.9)** | 83.58 **(8.58)**
 :sparkles: :new: | **app/client/src/utils/WidgetRegisterHelpers.tsx** | **100** | **100** | **100** | **100**
 :green_circle: | app/client/src/utils/WidgetRegistry.tsx | 100 **(20.51)** | 100 **(0)** | 100 **(61.54)** | 100 **(20.51)**
 :green_circle: | app/client/src/utils/storage.ts | 38.21 **(0)** | 13.64 **(0)** | 18.52 **(0)** | 31.87 **(0.76)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 39.13 **(0.87)** | 36.21 **(0)** | 55.35 **(0.27)**
 :green_circle: | app/client/src/utils/hooks/useDynamicAppLayout.tsx | 96.23 **(0.08)** | 56.52 **(0)** | 100 **(0)** | 95.74 **(0.09)**
 :red_circle: | app/client/src/utils/migrations/MenuButtonWidget.ts | 66.67 **(-2.56)** | 66.67 **(0)** | 100 **(0)** | 63.64 **(-3.03)**
 :red_circle: | app/client/src/utils/migrations/TextWidgetReplaceTextStyle.ts | 95.65 **(-0.18)** | 85.71 **(0)** | 100 **(0)** | 95.45 **(-0.2)**
 :red_circle: | app/client/src/widgets/BaseWidget.tsx | 74.59 **(-1.64)** | 39.53 **(-4.66)** | 68.75 **(-3.13)** | 73.5 **(-1.71)**
 :green_circle: | app/client/src/widgets/CanvasWidget.tsx | 95.92 **(2.59)** | 66.67 **(0)** | 100 **(12.5)** | 97.83 **(2.83)**
 :green_circle: | app/client/src/widgets/SkeletonWidget.tsx | 93.75 **(6.25)** | 100 **(0)** | 75 **(25)** | 93.33 **(6.66)**
 :sparkles: :new: | **app/client/src/widgets/WidgetUtils.ts** | **66.67** | **18.18** | **16.67** | **57.89**
 :sparkles: :new: | **app/client/src/widgets/constants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/AudioRecorderWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/AudioRecorderWidget/component/index.tsx** | **13.45** | **15.56** | **6.98** | **15.55**
 :sparkles: :new: | **app/client/src/widgets/AudioRecorderWidget/widget/index.tsx** | **47.37** | **0** | **62.5** | **47.22**
 :sparkles: :new: | **app/client/src/widgets/ButtonWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/ButtonWidget/component/index.tsx** | **45.05** | **21.39** | **36** | **42.86**
 :sparkles: :new: | **app/client/src/widgets/ButtonWidget/widget/index.tsx** | **65.31** | **37.5** | **60** | **64.58**
 :sparkles: :new: | **app/client/src/widgets/ChartWidget/constants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/ChartWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/ChartWidget/component/index.tsx** | **49.41** | **22.11** | **50** | **47.47**
 :sparkles: :new: | **app/client/src/widgets/ChartWidget/widget/index.tsx** | **95.83** | **66.67** | **88.89** | **95.45**
 :sparkles: :new: | **app/client/src/widgets/ChartWidget/widget/propertyConfig.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/CheckboxGroupWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/CheckboxGroupWidget/component/index.tsx** | **42.86** | **13.33** | **0** | **50**
 :sparkles: :new: | **app/client/src/widgets/CheckboxGroupWidget/widget/index.tsx** | **33.82** | **0** | **31.58** | **34.92**
 :sparkles: :new: | **app/client/src/widgets/CheckboxWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/CheckboxWidget/component/index.tsx** | **94.44** | **73.33** | **80** | **93.33**
 :sparkles: :new: | **app/client/src/widgets/CheckboxWidget/widget/index.tsx** | **95.83** | **66.67** | **87.5** | **95.45**
 :sparkles: :new: | **app/client/src/widgets/ContainerWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/ContainerWidget/component/index.tsx** | **93.1** | **65** | **100** | **93.1**
 :sparkles: :new: | **app/client/src/widgets/ContainerWidget/widget/index.tsx** | **98.31** | **77.78** | **100** | **98.21**
 :sparkles: :new: | **app/client/src/widgets/DatePickerWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/DatePickerWidget/component/index.tsx** | **27.03** | **2.9** | **6.25** | **26.39**
 :sparkles: :new: | **app/client/src/widgets/DatePickerWidget/widget/index.tsx** | **31.51** | **0** | **50** | **30.99**
 :sparkles: :new: | **app/client/src/widgets/DatePickerWidget2/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/DatePickerWidget2/component/index.tsx** | **23.81** | **2.99** | **5** | **23.75**
 :sparkles: :new: | **app/client/src/widgets/DatePickerWidget2/widget/index.tsx** | **84** | **0** | **75** | **86.96**
 :sparkles: :new: | **app/client/src/widgets/DividerWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/DividerWidget/component/index.tsx** | **64.58** | **42.25** | **41.67** | **69.05**
 :sparkles: :new: | **app/client/src/widgets/DividerWidget/widget/index.tsx** | **93.33** | **100** | **80** | **92.86**
 :sparkles: :new: | **app/client/src/widgets/DropdownWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/DropdownWidget/component/index.tsx** | **44.23** | **19.51** | **5** | **44**
 :sparkles: :new: | **app/client/src/widgets/DropdownWidget/widget/index.tsx** | **51.11** | **0** | **54.55** | **55**
 :sparkles: :new: | **app/client/src/widgets/FilePickerWidgetV2Widget/constants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/FilePickerWidgetV2Widget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/FilePickerWidgetV2Widget/component/index.tsx** | **52.38** | **0** | **20** | **50**
 :sparkles: :new: | **app/client/src/widgets/FilePickerWidgetV2Widget/widget/index.tsx** | **34.65** | **0** | **20.69** | **35.05**
 :sparkles: :new: | **app/client/src/widgets/FilepickerWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/FilepickerWidget/component/index.tsx** | **52.38** | **0** | **20** | **50**
 :sparkles: :new: | **app/client/src/widgets/FilepickerWidget/widget/FileDataTypes.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/FilepickerWidget/widget/index.tsx** | **38.37** | **0** | **21.43** | **37.65**
 :sparkles: :new: | **app/client/src/widgets/FormButtonWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/FormButtonWidget/widget/index.tsx** | **70.73** | **32** | **72.73** | **70**
 :sparkles: :new: | **app/client/src/widgets/FormWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/FormWidget/widget/index.tsx** | **87.76** | **52.63** | **92.31** | **88.89**
 :sparkles: :new: | **app/client/src/widgets/IconButtonWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/IconButtonWidget/component/index.tsx** | **31.25** | **4.08** | **0** | **43.48**
 :sparkles: :new: | **app/client/src/widgets/IconButtonWidget/widget/index.tsx** | **48.39** | **0** | **60** | **48.28**
 :sparkles: :new: | **app/client/src/widgets/IconWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/IconWidget/component/index.tsx** | **88.89** | **0** | **50** | **87.5**
 :sparkles: :new: | **app/client/src/widgets/IconWidget/widget/index.tsx** | **78.57** | **28.57** | **66.67** | **80.77**
 :sparkles: :new: | **app/client/src/widgets/IframeWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/IframeWidget/component/index.tsx** | **25** | **17.39** | **0** | **26.67**
 :sparkles: :new: | **app/client/src/widgets/IframeWidget/widget/index.tsx** | **48.48** | **0** | **57.14** | **48.39**
 :sparkles: :new: | **app/client/src/widgets/ImageWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/ImageWidget/component/index.tsx** | **15.84** | **15.63** | **8.33** | **16.67**
 :sparkles: :new: | **app/client/src/widgets/ImageWidget/widget/index.tsx** | **68.75** | **0** | **60** | **75**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/constants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/component/CurrencyCodeDropdown.tsx** | **60** | **40** | **40** | **58.33**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx** | **50** | **10** | **22.22** | **48.15**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/component/index.tsx** | **59.06** | **45.78** | **63.41** | **57.72**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/component/utilities.ts** | **100** | **50** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/InputWidget/widget/index.tsx** | **53.54** | **22.86** | **63.64** | **55.32**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/index.ts** | **9.09** | **0** | **0** | **10**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/component/ListPagination.tsx** | **66.67** | **33.33** | **0** | **66.67**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/component/index.tsx** | **80** | **50** | **50** | **80**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/widget/derived.js** | **62.26** | **35.71** | **50** | **60**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/widget/index.tsx** | **50.59** | **27.34** | **44.44** | **48.96**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/widget/parseDerivedProperties.ts** | **38.46** | **0** | **0** | **38.46**
 :sparkles: :new: | **app/client/src/widgets/ListWidget/widget/propertyConfig.ts** | **53.85** | **0** | **0** | **53.85**
 :sparkles: :new: | **app/client/src/widgets/MapWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/MapWidget/component/PickMyLocation.tsx** | **37.5** | **25** | **16.67** | **38.1**
 :sparkles: :new: | **app/client/src/widgets/MapWidget/component/index.tsx** | **30.23** | **24.24** | **0** | **31.71**
 :sparkles: :new: | **app/client/src/widgets/MapWidget/widget/index.tsx** | **48.39** | **9.09** | **33.33** | **48.33**
 :sparkles: :new: | **app/client/src/widgets/MenuButtonWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/MenuButtonWidget/component/index.tsx** | **24.64** | **6.62** | **0** | **27.87**
 :sparkles: :new: | **app/client/src/widgets/MenuButtonWidget/widget/index.tsx** | **47.37** | **0** | **36.36** | **47.22**
 :sparkles: :new: | **app/client/src/widgets/ModalWidget/index.ts** | **38.46** | **0** | **0** | **38.46**
 :sparkles: :new: | **app/client/src/widgets/ModalWidget/component/index.tsx** | **36.36** | **22.22** | **0** | **36.36**
 :sparkles: :new: | **app/client/src/widgets/ModalWidget/widget/index.tsx** | **43.48** | **0** | **20** | **43.08**
 :sparkles: :new: | **app/client/src/widgets/MultiSelectWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/MultiSelectWidget/component/index.styled.tsx** | **92.31** | **100** | **0** | **100**
 :sparkles: :new: | **app/client/src/widgets/MultiSelectWidget/component/index.tsx** | **17.31** | **0** | **0** | **18.75**
 :sparkles: :new: | **app/client/src/widgets/MultiSelectWidget/widget/index.tsx** | **50** | **0** | **50** | **51.11**
 :sparkles: :new: | **app/client/src/widgets/RadioGroupWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/RadioGroupWidget/component/index.tsx** | **66.67** | **30.77** | **25** | **73.33**
 :sparkles: :new: | **app/client/src/widgets/RadioGroupWidget/widget/index.tsx** | **84** | **0** | **75** | **86.96**
 :sparkles: :new: | **app/client/src/widgets/RateWidget/constants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/RateWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/RateWidget/component/index.tsx** | **32.43** | **18.18** | **0** | **37.5**
 :sparkles: :new: | **app/client/src/widgets/RateWidget/widget/index.tsx** | **51.22** | **0** | **66.67** | **51.28**
 :sparkles: :new: | **app/client/src/widgets/RichTextEditorWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/RichTextEditorWidget/widget/index.tsx** | **72.22** | **22.22** | **58.33** | **73.53**
 :sparkles: :new: | **app/client/src/widgets/StatboxWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/StatboxWidget/widget/index.tsx** | **68.75** | **0** | **60** | **66.67**
 :sparkles: :new: | **app/client/src/widgets/SwitchWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/SwitchWidget/component/index.tsx** | **93.33** | **62.5** | **50** | **93.33**
 :sparkles: :new: | **app/client/src/widgets/SwitchWidget/widget/index.tsx** | **95.83** | **60** | **87.5** | **95.45**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/constants.ts** | **100** | **50** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/index.ts** | **50** | **100** | **0** | **50**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx** | **27.27** | **14.29** | **0** | **31.03**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/CascadeFields.tsx** | **28.13** | **24.66** | **3.13** | **29.75**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/CommonUtilities.ts** | **88.57** | **48.94** | **100** | **87.5**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/Constants.ts** | **85.48** | **73.53** | **91.3** | **85.48**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/Table.tsx** | **61.63** | **38.89** | **38.89** | **60.98**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableAction.tsx** | **42.86** | **16.67** | **0** | **42.86**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableDataDownload.tsx** | **35.82** | **13.89** | **9.09** | **34.92**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableFilterPane.tsx** | **60.98** | **22.22** | **12.5** | **61.54**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableFilterPaneContent.tsx** | **30.19** | **29.27** | **0** | **31.37**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableFilters.tsx** | **78.38** | **23.81** | **66.67** | **77.14**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableHeader.tsx** | **48.72** | **51.06** | **30** | **45.95**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableHelpers.ts** | **100** | **75** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableStyledWrappers.tsx** | **67.82** | **20.96** | **50** | **67.82**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/TableUtilities.tsx** | **39.86** | **23.3** | **15.79** | **38.46**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/component/index.tsx** | **27.27** | **2.63** | **15.79** | **25.68**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/widget/derived.js** | **54.5** | **53.57** | **39.47** | **54.07**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/widget/index.tsx** | **26.73** | **5.98** | **26.42** | **26.63**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/widget/parseDerivedProperties.ts** | **38.46** | **0** | **0** | **38.46**
 :sparkles: :new: | **app/client/src/widgets/TableWidget/widget/propertyConfig.ts** | **38.73** | **10.2** | **65.63** | **40.15**
 :sparkles: :new: | **app/client/src/widgets/TabsMigrator/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TabsMigrator/widget/index.tsx** | **78.26** | **0** | **60** | **77.27**
 :sparkles: :new: | **app/client/src/widgets/TabsWidget/index.ts** | **42.86** | **100** | **0** | **42.86**
 :sparkles: :new: | **app/client/src/widgets/TabsWidget/component/index.tsx** | **100** | **83.33** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TabsWidget/widget/index.tsx** | **67.15** | **45.54** | **60** | **66.67**
 :sparkles: :new: | **app/client/src/widgets/TextWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TextWidget/component/index.tsx** | **100** | **72.41** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TextWidget/widget/index.tsx** | **100** | **50** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/VideoWidget/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/VideoWidget/component/PopoverVideo.tsx** | **81.82** | **100** | **0** | **81.82**
 :sparkles: :new: | **app/client/src/widgets/VideoWidget/component/index.tsx** | **77.78** | **50** | **0** | **77.78**
 :sparkles: :new: | **app/client/src/widgets/VideoWidget/widget/index.tsx** | **63.41** | **40** | **46.15** | **80.65**
 :x: | ~~app/client/src/actions/widgetSidebarActions.tsx~~ | ~~72.73~~ | ~~100~~ | ~~0~~ | ~~62.5~~
 :x: | ~~app/client/src/api/WidgetSidebarApi.tsx~~ | ~~87.5~~ | ~~100~~ | ~~50~~ | ~~85.71~~
 :x: | ~~app/client/src/components/ads/CurrencyCodeDropdown.tsx~~ | ~~60~~ | ~~40~~ | ~~40~~ | ~~58.33~~
 :x: | ~~app/client/src/components/ads/ISDCodeDropdown.tsx~~ | ~~50~~ | ~~10~~ | ~~22.22~~ | ~~48.15~~
 :x: | ~~app/client/src/components/designSystems/appsmith/AudioRecorderComponent.tsx~~ | ~~13.45~~ | ~~15.56~~ | ~~6.98~~ | ~~15.55~~
 :x: | ~~app/client/src/components/designSystems/appsmith/ChartComponent.tsx~~ | ~~49.41~~ | ~~22.11~~ | ~~50~~ | ~~47.47~~
 :x: | ~~app/client/src/components/designSystems/appsmith/CheckboxGroupComponent.tsx~~ | ~~42.86~~ | ~~13.33~~ | ~~0~~ | ~~50~~
 :x: | ~~app/client/src/components/designSystems/appsmith/ContainerComponent.tsx~~ | ~~93.1~~ | ~~65~~ | ~~100~~ | ~~93.1~~
 :x: | ~~app/client/src/components/designSystems/appsmith/FilePickerComponent.tsx~~ | ~~52.38~~ | ~~0~~ | ~~20~~ | ~~50~~
 :x: | ~~app/client/src/components/designSystems/appsmith/IconButtonComponent.tsx~~ | ~~48.84~~ | ~~7.84~~ | ~~20~~ | ~~61.76~~
 :x: | ~~app/client/src/components/designSystems/appsmith/IconComponent.tsx~~ | ~~88.89~~ | ~~0~~ | ~~50~~ | ~~87.5~~
 :x: | ~~app/client/src/components/designSystems/appsmith/ImageComponent.tsx~~ | ~~15.84~~ | ~~15.63~~ | ~~8.33~~ | ~~16.67~~
 :x: | ~~app/client/src/components/designSystems/appsmith/MapComponent.tsx~~ | ~~30.23~~ | ~~24.24~~ | ~~0~~ | ~~31.71~~
 :x: | ~~app/client/src/components/designSystems/appsmith/PickMyLocation.tsx~~ | ~~37.5~~ | ~~25~~ | ~~16.67~~ | ~~38.1~~
 :x: | ~~app/client/src/components/designSystems/appsmith/PopoverVideo.tsx~~ | ~~81.82~~ | ~~100~~ | ~~0~~ | ~~81.82~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TabsComponent.tsx~~ | ~~100~~ | ~~83.33~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/components/designSystems/appsmith/VideoComponent.tsx~~ | ~~77.78~~ | ~~50~~ | ~~0~~ | ~~77.78~~
 :x: | ~~app/client/src/components/designSystems/appsmith/MenuButtonComponent/index.tsx~~ | ~~26.76~~ | ~~6.62~~ | ~~0~~ | ~~30.16~~
 :x: | ~~app/client/src/components/designSystems/appsmith/MultiSelectComponent/index.styled.tsx~~ | ~~92.31~~ | ~~100~~ | ~~0~~ | ~~100~~
 :x: | ~~app/client/src/components/designSystems/appsmith/MultiSelectComponent/index.tsx~~ | ~~17.31~~ | ~~0~~ | ~~0~~ | ~~18.75~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/AutoToolTipComponent.tsx~~ | ~~27.27~~ | ~~14.29~~ | ~~0~~ | ~~31.03~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/CascadeFields.tsx~~ | ~~28.13~~ | ~~24.66~~ | ~~3.13~~ | ~~29.75~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/CommonUtilities.ts~~ | ~~88.57~~ | ~~48.94~~ | ~~100~~ | ~~87.5~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/Constants.ts~~ | ~~85.48~~ | ~~73.53~~ | ~~91.3~~ | ~~85.48~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/Table.tsx~~ | ~~61.63~~ | ~~38.89~~ | ~~38.89~~ | ~~60.98~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableAction.tsx~~ | ~~71.43~~ | ~~58.33~~ | ~~66.67~~ | ~~71.43~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableDataDownload.tsx~~ | ~~34.85~~ | ~~13.89~~ | ~~9.09~~ | ~~33.87~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableFilterPane.tsx~~ | ~~60.98~~ | ~~22.22~~ | ~~12.5~~ | ~~61.54~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableFilterPaneContent.tsx~~ | ~~30.19~~ | ~~29.27~~ | ~~0~~ | ~~31.37~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableFilters.tsx~~ | ~~78.38~~ | ~~23.81~~ | ~~66.67~~ | ~~77.14~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx~~ | ~~48.72~~ | ~~51.06~~ | ~~30~~ | ~~45.95~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableHelpers.ts~~ | ~~100~~ | ~~75~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableStyledWrappers.tsx~~ | ~~67.82~~ | ~~20.96~~ | ~~50~~ | ~~67.82~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx~~ | ~~39.86~~ | ~~23.3~~ | ~~15.79~~ | ~~38.46~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/index.tsx~~ | ~~27.27~~ | ~~2.63~~ | ~~15.79~~ | ~~25.68~~
 :x: | ~~app/client/src/components/designSystems/blueprint/ButtonComponent.tsx~~ | ~~50.41~~ | ~~23.04~~ | ~~40.74~~ | ~~49.09~~
 :x: | ~~app/client/src/components/designSystems/blueprint/CheckboxComponent.tsx~~ | ~~94.44~~ | ~~73.33~~ | ~~80~~ | ~~93.33~~
 :x: | ~~app/client/src/components/designSystems/blueprint/CloseButton.tsx~~ | ~~55.56~~ | ~~100~~ | ~~0~~ | ~~55.56~~
 :x: | ~~app/client/src/components/designSystems/blueprint/DatePickerComponent.tsx~~ | ~~27.03~~ | ~~2.9~~ | ~~6.25~~ | ~~26.39~~
 :x: | ~~app/client/src/components/designSystems/blueprint/DatePickerComponent2.tsx~~ | ~~23.81~~ | ~~2.99~~ | ~~5~~ | ~~23.75~~
 :x: | ~~app/client/src/components/designSystems/blueprint/DividerComponent.tsx~~ | ~~64.58~~ | ~~42.25~~ | ~~41.67~~ | ~~69.05~~
 :x: | ~~app/client/src/components/designSystems/blueprint/DropdownComponent.tsx~~ | ~~44.23~~ | ~~19.51~~ | ~~5~~ | ~~44~~
 :x: | ~~app/client/src/components/designSystems/blueprint/IframeComponent.tsx~~ | ~~25~~ | ~~17.39~~ | ~~0~~ | ~~26.67~~
 :x: | ~~app/client/src/components/designSystems/blueprint/ModalComponent.tsx~~ | ~~33.33~~ | ~~15.38~~ | ~~0~~ | ~~34.48~~
 :x: | ~~app/client/src/components/designSystems/blueprint/RadioGroupComponent.tsx~~ | ~~66.67~~ | ~~30.77~~ | ~~25~~ | ~~73.33~~
 :x: | ~~app/client/src/components/designSystems/blueprint/RateComponent.tsx~~ | ~~32.43~~ | ~~18.18~~ | ~~0~~ | ~~37.5~~
 :x: | ~~app/client/src/components/designSystems/blueprint/SwitchComponent.tsx~~ | ~~93.33~~ | ~~62.5~~ | ~~50~~ | ~~93.33~~
 :x: | ~~app/client/src/components/designSystems/blueprint/TextComponent.tsx~~ | ~~100~~ | ~~73.33~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/components/designSystems/blueprint/InputComponent/index.tsx~~ | ~~59.06~~ | ~~45.78~~ | ~~63.41~~ | ~~57.72~~
 :x: | ~~app/client/src/components/designSystems/blueprint/InputComponent/utilties.tsx~~ | ~~100~~ | ~~50~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/constants/ChartConstants.ts~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/mockResponses/WidgetConfigResponse.tsx~~ | ~~14.81~~ | ~~0~~ | ~~0~~ | ~~15.84~~
 :x: | ~~app/client/src/mockResponses/WidgetSidebarResponse.tsx~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/reducers/uiReducers/widgetSidebarReducer.ts~~ | ~~83.33~~ | ~~100~~ | ~~0~~ | ~~83.33~~
 :x: | ~~app/client/src/sagas/WidgetBlueprintSagasEnums.ts~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/sagas/WidgetSidebarSagas.tsx~~ | ~~63.64~~ | ~~100~~ | ~~50~~ | ~~70~~
 :x: | ~~app/client/src/widgets/AudioRecorderWidget.tsx~~ | ~~47.62~~ | ~~0~~ | ~~50~~ | ~~47.5~~
 :x: | ~~app/client/src/widgets/ButtonWidget.tsx~~ | ~~64~~ | ~~37.5~~ | ~~50~~ | ~~63.27~~
 :x: | ~~app/client/src/widgets/CheckboxGroupWidget.tsx~~ | ~~35.21~~ | ~~0~~ | ~~26.32~~ | ~~36.36~~
 :x: | ~~app/client/src/widgets/CheckboxWidget.tsx~~ | ~~92.59~~ | ~~66.67~~ | ~~75~~ | ~~92~~
 :x: | ~~app/client/src/widgets/ContainerWidget.tsx~~ | ~~96.43~~ | ~~77.78~~ | ~~90~~ | ~~96.23~~
 :x: | ~~app/client/src/widgets/DatePickerWidget.tsx~~ | ~~32.89~~ | ~~0~~ | ~~41.67~~ | ~~32.43~~
 :x: | ~~app/client/src/widgets/DatePickerWidget2.tsx~~ | ~~82.14~~ | ~~0~~ | ~~62.5~~ | ~~84.62~~
 :x: | ~~app/client/src/widgets/DividerWidget.tsx~~ | ~~88.24~~ | ~~100~~ | ~~60~~ | ~~87.5~~
 :x: | ~~app/client/src/widgets/DropdownWidget.tsx~~ | ~~52.08~~ | ~~0~~ | ~~45.45~~ | ~~55.81~~
 :x: | ~~app/client/src/widgets/FileDataTypes.ts~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/widgets/FilepickerWidget.tsx~~ | ~~38.46~~ | ~~0~~ | ~~17.86~~ | ~~37.78~~
 :x: | ~~app/client/src/widgets/FilepickerWidgetV2.tsx~~ | ~~35.58~~ | ~~0~~ | ~~17.24~~ | ~~36~~
 :x: | ~~app/client/src/widgets/FormButtonWidget.tsx~~ | ~~67.5~~ | ~~32~~ | ~~55.56~~ | ~~66.67~~
 :x: | ~~app/client/src/widgets/FormWidget.tsx~~ | ~~86.27~~ | ~~52.63~~ | ~~84.62~~ | ~~87.23~~
 :x: | ~~app/client/src/widgets/IconButtonWidget.tsx~~ | ~~50~~ | ~~0~~ | ~~40~~ | ~~50~~
 :x: | ~~app/client/src/widgets/IconWidget.tsx~~ | ~~72~~ | ~~28.57~~ | ~~33.33~~ | ~~73.91~~
 :x: | ~~app/client/src/widgets/IframeWidget.tsx~~ | ~~51.35~~ | ~~0~~ | ~~42.86~~ | ~~51.43~~
 :x: | ~~app/client/src/widgets/ImageWidget.tsx~~ | ~~60.71~~ | ~~0~~ | ~~28.57~~ | ~~66.67~~
 :x: | ~~app/client/src/widgets/InputWidget.tsx~~ | ~~53.92~~ | ~~22.86~~ | ~~59.09~~ | ~~55.67~~
 :x: | ~~app/client/src/widgets/MapWidget.tsx~~ | ~~47.62~~ | ~~9.09~~ | ~~23.53~~ | ~~48.33~~
 :x: | ~~app/client/src/widgets/MenuButtonWidget.tsx~~ | ~~47.5~~ | ~~0~~ | ~~27.27~~ | ~~47.37~~
 :x: | ~~app/client/src/widgets/ModalWidget.tsx~~ | ~~45.07~~ | ~~0~~ | ~~13.33~~ | ~~44.78~~
 :x: | ~~app/client/src/widgets/MultiSelectWidget.tsx~~ | ~~50.98~~ | ~~0~~ | ~~41.67~~ | ~~52.08~~
 :x: | ~~app/client/src/widgets/RadioGroupWidget.tsx~~ | ~~82.14~~ | ~~0~~ | ~~62.5~~ | ~~84.62~~
 :x: | ~~app/client/src/widgets/RichTextEditorWidget.tsx~~ | ~~71.79~~ | ~~22.22~~ | ~~50~~ | ~~72.97~~
 :x: | ~~app/client/src/widgets/StatboxWidget.tsx~~ | ~~68.42~~ | ~~0~~ | ~~40~~ | ~~66.67~~
 :x: | ~~app/client/src/widgets/SwitchWidget.tsx~~ | ~~92.59~~ | ~~60~~ | ~~75~~ | ~~92~~
 :x: | ~~app/client/src/widgets/TextWidget.tsx~~ | ~~94.44~~ | ~~50~~ | ~~80~~ | ~~94.12~~
 :x: | ~~app/client/src/widgets/VideoWidget.tsx~~ | ~~63.64~~ | ~~40~~ | ~~38.46~~ | ~~75~~
 :x: | ~~app/client/src/widgets/ChartWidget/index.tsx~~ | ~~92.59~~ | ~~66.67~~ | ~~77.78~~ | ~~92~~
 :x: | ~~app/client/src/widgets/ChartWidget/propertyConfig.ts~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/widgets/ListWidget/ListComponent.tsx~~ | ~~93.33~~ | ~~66.67~~ | ~~83.33~~ | ~~93.33~~
 :x: | ~~app/client/src/widgets/ListWidget/ListPagination.tsx~~ | ~~66.67~~ | ~~33.33~~ | ~~0~~ | ~~66.67~~
 :x: | ~~app/client/src/widgets/ListWidget/ListPropertyPaneConfig.ts~~ | ~~53.85~~ | ~~0~~ | ~~0~~ | ~~53.85~~
 :x: | ~~app/client/src/widgets/ListWidget/ListWidget.tsx~~ | ~~52.24~~ | ~~33.12~~ | ~~42.55~~ | ~~50.79~~
 :x: | ~~app/client/src/widgets/ListWidget/derived.js~~ | ~~66~~ | ~~41.67~~ | ~~62.5~~ | ~~63.83~~
 :x: | ~~app/client/src/widgets/ListWidget/parseDerivedProperties.ts~~ | ~~38.46~~ | ~~0~~ | ~~0~~ | ~~38.46~~
 :x: | ~~app/client/src/widgets/RateWidget/index.tsx~~ | ~~52.27~~ | ~~0~~ | ~~55.56~~ | ~~52.38~~
 :x: | ~~app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts~~ | ~~38.73~~ | ~~10.2~~ | ~~65.63~~ | ~~40.15~~
 :x: | ~~app/client/src/widgets/TableWidget/TableWidgetConstants.ts~~ | ~~100~~ | ~~50~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/widgets/TableWidget/derived.js~~ | ~~54.5~~ | ~~53.57~~ | ~~39.47~~ | ~~54.07~~
 :x: | ~~app/client/src/widgets/TableWidget/index.tsx~~ | ~~27.09~~ | ~~5.98~~ | ~~25~~ | ~~27.22~~
 :x: | ~~app/client/src/widgets/TableWidget/parseDerivedProperties.ts~~ | ~~38.46~~ | ~~0~~ | ~~0~~ | ~~38.46~~
 :x: | ~~app/client/src/widgets/Tabs/TabsMigrator.tsx~~ | ~~77.78~~ | ~~0~~ | ~~40~~ | ~~76.92~~
 :x: | ~~app/client/src/widgets/Tabs/TabsWidget.tsx~~ | ~~67.86~~ | ~~44.33~~ | ~~56.67~~ | ~~67.42~~</details>

riodeuno commented on app/client/src/pages/AppViewer/AppPage.tsx:
```diff
@@ -19,6 +18,7 @@ type AppPageProps = {
   pageName?: string;
   pageId?: string;
   appName?: string;
+  width: number;
```
This is redundant. Remove.

riodeuno commented on app/client/src/pages/AppViewer/AppViewerPageContainer.tsx:
```diff
@@ -40,6 +41,7 @@ type AppViewerPageContainerProps = {
   currentAppName?: string;
   fetchPage: (pageId: string, bustCache?: boolean) => void;
   currentAppPermissions?: string[];
+  width: number;
```
This is not needed anymore. Remove.

riodeuno commented on app/client/src/pages/Editor/Canvas.tsx:
```diff
@@ -1,41 +1,50 @@
-            WidgetFactory.createWidget(props.dsl, RenderModes.CANVAS)}
-        </ArtBoard>
-      </>
+      prevProps.width === nextProps.width &&
```
Remove this, as this is not needed anymore.

riodeuno commented on app/client/src/widgets/DatePickerWidget2/widget/index.tsx:
```diff
@@ -1,13 +1,14 @@
 import React from "react";
-import BaseWidget, { WidgetProps, WidgetState } from "./BaseWidget";
+import BaseWidget, { WidgetProps, WidgetState } from "../../BaseWidget";
```
Absolute imports please.

riodeuno commented on app/client/src/constants/HelpConstants.ts:
```diff
@@ -1,4 +1,4 @@
-export const HelpMap = {
+export const HelpMap: Record<string, { path: string; searchKey: string }> = {
```
Uses WidgetTypes as keys, which are not statically available, as the widgets are registered dynamically.

riodeuno commented on app/client/src/sagas/WidgetOperationSagas.tsx:
```diff
@@ -470,7 +487,7 @@ export function* deleteAllSelectedWidgetsSaga(
 ) {
   try {
-    const { disallowUndo = false, isShortcut } = deleteAction.payload;
+    const { disallowUndo = false } = deleteAction.payload;
```
unused variable.

riodeuno commented on app/client/src/utils/autocomplete/EntityDefinitions.ts:
```diff
@@ -7,7 +7,7 @@ const isVisible = {
 };
 
-export const entityDefinitions = {
+export const entityDefinitions: Record<string, unknown> = {
```
Same issue as HelpConstants

riodeuno commented on app/client/src/components/propertyControls/ChartDataControl.tsx:
```diff
@@ -15,7 +15,7 @@ import {
 } from "components/editorComponents/CodeEditor/EditorConfig";
 import { Size, Category } from "components/ads/Button";
-import { AllChartData, ChartData } from "widgets/ChartWidget";
+import { AllChartData, ChartData } from "widgets/ChartWidget/constants";
```
This leaks should be fixed by creating a property control architecture, which does not require widget specific types.

hetunandu commented on app/client/src/sagas/DebuggerSagas.ts:
```diff
@@ -316,6 +332,7 @@ function* logDebuggerErrorAnalyticsSaga(
 export default function* debuggerSagasListeners() {
   yield all([
     takeEvery(ReduxActionTypes.DEBUGGER_LOG_INIT, debuggerLogSaga),
+    takeEvery(ReduxActionTypes.EXECUTE_ACTION, logActionSaga),
```
LGTM
cc @akash-codemonk

riodeuno commented on app/client/src/widgets/ChartWidget/widget/propertyConfig.ts:
```diff
@@ -1,4 +1,4 @@
-import { ChartWidgetProps } from "widgets/ChartWidget";
+import { ChartWidgetProps } from "widgets/ChartWidget/widget";
```
Not necessary at this point.

riodeuno commented:
/ok-to-test sha=6f4d811

riodeuno commented:
/ok-to-test sha=f5e8097

riodeuno commented:
/ok-to-test sha=45c2d5d

somangshu reviewed (dismissed):
Approving to run tests

vivekverma2312 commented:
@riodeuno 1. Added tab in Tab widget is not showing properly in entity explorer.

<img width="1295" alt="Screenshot 2021-08-31 at 8 46 24 PM" src="https://user-images.githubusercontent.com/83569920/131530266-220b2ea4-767d-40f2-88cd-7fd7c774f7ee.png">

YogeshJayaseelan commented:
@riodeuno issue 2 - On button widget setting up onClick actions to navigate to 2 page and coming back to original page the page breaks and not able to perform any actions on page 1. 
Note: This onClick settings was made 2 days ago and did not experience this issue earlier and this is happening from this morning (02/09/2021)

https://www.loom.com/share/220e8254e84b47b4b5677b04f4a458fb

riodeuno commented:
@YogeshJayaseelan Thanks for pointing out the timelines for this issue. Confirming a few things
- This issue did not exist in the PR on 31st and 1st. 
- The issue cropped up on 2nd. 
- The issue does not exist on release.app.appsmith.com

This leads me to believe that this issue came about in the release merge on 1st afternoon. I'll look into it.

YogeshJayaseelan commented:
> @YogeshJayaseelan Thanks for pointing out the timelines for this issue. Confirming a few things
> 
> * This issue did not exist in the PR on 31st and 1st.
> * The issue cropped up on 2nd.
> * The issue does not exist on release.app.appsmith.com
> 
> This leads me to believe that this issue came about in the release merge on 1st afternoon. I'll look into it.

Yes this might be due to release merge conflicts

YogeshJayaseelan commented:
@riodeuno issue 3 - Iframe - onUrlChanged setting navigate to invalid page (Page2 - Which is actually not created) and changed the url and it navigates to invalid page and coming back to the original page again the page loops back to the invalid page and not able to access the app 

https://www.loom.com/share/21bcef6eacfc405188646fa221bbd885

YogeshJayaseelan commented:
@riodeuno issue 4 - issue #7002 exists in this PR - It is fixed on release last week and this is not happening in release and prod now

YogeshJayaseelan commented:
@riodeuno issue 5 - Input widget - PhoneNumber Type - OnSubmit action - Entering phone number and clicking on enter button the phone number disappears. This is not happening on release 

https://www.loom.com/share/b6ba40a7b4af47f685af4ee15cb13555

YogeshJayaseelan commented:
@riodeuno The below issues for observed 'Once Thus far' and we are not able to reproduce 

6. Renaming List widget the widget becomes blank
7. Page no showing is 382 for default data
8. List widget overlaps Modal widget - Modal hide behind list widget

riodeuno commented:
> issue 5 - Input widget - PhoneNumber Type - OnSubmit action - Entering phone number and clicking on enter button the phone number disappears. This is not happening on release and in prod phone number feature is not implemented yet
> 
> The same behaviour for currency type as well - This is not happening in release and happens in prod
> 

@YogeshJayaseelan I can see this happening in release as well

riodeuno commented:
> @riodeuno issue 3 - Iframe - onUrlChanged setting navigate to invalid page (Page2 - Which is actually not created) and changed the url and it navigates to invalid page and coming back to the original page again the page loops back to the invalid page and not able to access the app
> 
> [![LOOM DEMO](https://camo.githubusercontent.com/2f45ea5a8e6c971531c4fa3ec0344f0fedb4b4dfdd3375010af85bc8d83dd993/687474703a2f2f63646e2e6c6f6f6d2e636f6d2f73657373696f6e732f7468756d626e61696c732f32316263656636656163666334303531383836343666613232316262643838352d30303030312e676966)](https://www.loom.com/share/21bcef6eacfc405188646fa221bbd885)

I couldn't replicate this today. Assuming this to be fixed. As the code around this feature has remained unedited. We can revisit if we're able to replicate this issue again.

riodeuno commented:
> @riodeuno The below issues for observed 'Once Thus far' and we are not able to reproduce
> 
> 1. Renaming List widget the widget becomes blank
> 2. Page no showing is 382 for default data
> 3. List widget overlaps Modal widget - Modal hide behind list widget

@YogeshJayaseelan I couldn't replicate any of these issues.

riodeuno commented:
> @riodeuno issue 4 - issue #7002 exists in this PR - It is fixed on release last week and this is not happening in release and prod now

@YogeshJayaseelan I've fixed this. Was a matter of code which wasn't moved earlier.

riodeuno commented:
> @riodeuno issue 2 - On button widget setting up onClick actions to navigate to 2 page and coming back to original page the page breaks and not able to perform any actions on page 1.
> Note: This onClick settings was made 2 days ago and did not experience this issue earlier and this is happening from this morning (02/09/2021)
> 
> [![LOOM DEMO](https://camo.githubusercontent.com/e3bab4c2815bac199ad9611c522ad562bb5b20b62a9879e1ce5c807b5425a2d7/687474703a2f2f63646e2e6c6f6f6d2e636f6d2f73657373696f6e732f7468756d626e61696c732f32323065383235346538346234376234623536373762303466346134353866622d30303030312e676966)](https://www.loom.com/share/220e8254e84b47b4b5677b04f4a458fb)

@YogeshJayaseelan I can't replicate this issue as well.

riodeuno commented:
> @riodeuno 1. Added tab in Tab widget is not showing properly in entity explorer.
> 
> <img alt="Screenshot 2021-08-31 at 8 46 24 PM" width="1295" src="https://user-images.githubusercontent.com/83569920/131530266-220b2ea4-767d-40f2-88cd-7fd7c774f7ee.png">

@vivekverma2312 This was a legitimate issue. Nice find! I've fixed this.

Tooluloope reviewed (approved):
Approve For testing
